### ISSUE TITLE
Error in newly initialized git repo.

### ISSUE THREAD

**Issue by Nageswari-droid** [neel1996/gitconvex#155]
**Describe the bug**
When a new git repo is added, stage, commit, push operations are not available for these repositories.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to repositories
2. Click on the newly initialized git repo
3. Try to stage, commit or push
4. See an error

**Expected behaviour**
It should have all the basic operations.

**Screenshots**
![newGitRepo](https://user-images.githubusercontent.com/65342122/114582370-25ed3300-9c9e-11eb-93d1-a104665d2f30.PNG)

**Desktop (please complete the following information):**
 - OS : Windows
 - Browser : Chrome
 - Version : v2.1.0

**Issue Type**
>mark with [x] to select the check box

- [ ] UI Issue
- [x] Performance Issue
- [x] Feature Breaking Issue
- [ ] Security Issue

**Severity**
>mark with [x] to select the check box

- [x] Critical (The application crashes / breaks completely)
- [ ] High (Blocks you from performing a critical operation)
- [ ] Moderate (Bearable or Manageable issue) 
- [ ] Low (Something is not right, but the impact is minimal)

**Comment by Nageswari-droid** [comment]
```
2021/04/20 13:42:19 [WARNING]: Repo has no HEAD. Proceeding with a NIL HEAD
2021/04/20 13:42:19 [ERROR]: goroutine 34 [running]:
runtime/debug.Stack(0xc00009eb78, 0xdb7260, 0xd30f40)
        C:/hostedtoolcache/windows/go/1.16.3/x64/src/runtime/debug/stack.go:24 +0xa5
github.com/neel1996/gitconvex-server/git.CommitStruct.CommitChanges.func1(0x0, 0x0)
        D:/a/gitconvex/gitconvex/git/git_commit_changes.go:41 +0x45
panic(0xdb7260, 0xd30f40)
        C:/hostedtoolcache/windows/go/1.16.3/x64/src/runtime/panic.go:965 +0x1c7
github.com/libgit2/git2go/v31.(*Repository).CreateCommit(0xc0000f2d40, 0xe42bd7, 0x4, 0xc0000f2d80, 0xc0000f2d80, 0xc00017e254, 0xb, 0xc000498660, 0xc00009f090, 0x1, ...)
        C:/Users/runneradmin/go/pkg/mod/github.com/libgit2/git2go/v31@v31.4.10/repository.go:453 +0x142
github.com/neel1996/gitconvex-server/git.CommitStruct.CommitChanges(0xc0000f2d40, 0xc00017e254, 0xb, 0xc00018c420, 0x2f, 0x0, 0x0)
        D:/a/gitconvex/gitconvex/git/git_commit_changes.go:87 +0x2f8
github.com/neel1996/gitconvex-server/graph.(*mutationResolver).CommitChanges(0xc0000d8138, 0xeeed80, 0xc000480cc0, 0xc00017e239, 0x8, 0xc00017e254, 0xb, 0xdb2a60, 0xc000480d20, 0x193ffce05e0, ...)
        D:/a/gitconvex/gitconvex/graph/schema.resolvers.go:265 +0x1da
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Mutation_commitChanges.func2(0xeeed80, 0xc000480cc0, 0x20, 0xdfabc0, 0xc000089601, 0xc0000a6a80)
        D:/a/gitconvex/gitconvex/graph/generated/generated.go:3744 +0x19c
github.com/99designs/gqlgen/graphql/executor.processExtensions.func3(0xeeed80, 0xc000480cc0, 0xc0000a6a80, 0xc000480cf0, 0xc000480d20, 0x0, 0x0)
        C:/Users/runneradmin/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/extensions.go:62 +0x42
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Mutation_commitChanges(0xc0000895e0, 0xeeed80, 0xc000480c30, 0xc000186880, 0x0, 0x0, 0x0, 0x0, 0x0)
        D:/a/gitconvex/gitconvex/graph/generated/generated.go:3742 +0x2d3
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Mutation(0xc0000895e0, 0xeeed80, 0xc000480c30, 0xc000089370, 0x1, 0x1, 0x193800235a8, 0xc00009f6d8)
        D:/a/gitconvex/gitconvex/graph/generated/generated.go:7101 +0xec6
github.com/neel1996/gitconvex-server/graph/generated.(*executableSchema).Exec.func2(0xeeed80, 0xc000480c00, 0x30)
        D:/a/gitconvex/gitconvex/graph/generated/generated.go:1032 +0x85
github.com/99designs/gqlgen/graphql/executor.(*Executor).DispatchOperation.func1.1.1(0xeeed80, 0xc000480c00, 0xc0000895f0)
        C:/Users/runneradmin/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/executor.go:105 +0x4a
github.com/99designs/gqlgen/graphql/executor.processExtensions.func2(0xeeed80, 0xc000480c00, 0xc0000895f0, 0xed85c0)
        C:/Users/runneradmin/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/extensions.go:59 +0x42
github.com/99designs/gqlgen/graphql/executor.(*Executor).DispatchOperation.func1.1(0xeeed80, 0xc000480b70, 0xc000480b40)
        C:/Users/runneradmin/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/executor.go:104 +0x131
github.com/99designs/gqlgen/graphql/handler/transport.POST.Do(0xeede90, 0xc0002880e0, 0xc00017aa00, 0xeed860, 0xc0001b02d0)
        C:/Users/runneradmin/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/handler/transport/http_post.go:53 +0x3a8
github.com/99designs/gqlgen/graphql/handler.(*Server).ServeHTTP(0xc0001a2bc0, 0xeede90, 0xc0002880e0, 0xc00017aa00)
        C:/Users/runneradmin/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/handler/server.go:115 +0x21b
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0001d1b00, 0xeede90, 0xc0002880e0, 0xc00017a600)
        C:/Users/runneradmin/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0xe5
github.com/rs/cors.(*Cors).Handler.func1(0xeede90, 0xc0002880e0, 0xc00017a600)
        C:/Users/runneradmin/go/pkg/mod/github.com/rs/cors@v1.7.0/cors.go:219 +0x1c6
net/http.HandlerFunc.ServeHTTP(0xc0001a2de0, 0xeede90, 0xc0002880e0, 0xc00017a600)
        C:/hostedtoolcache/windows/go/1.16.3/x64/src/net/http/server.go:2069 +0x4b
net/http.serverHandler.ServeHTTP(0xc000210000, 0xeede90, 0xc0002880e0, 0xc00017a600)
        C:/hostedtoolcache/windows/go/1.16.3/x64/src/net/http/server.go:2887 +0xaa
net/http.(*conn).serve(0xc000200b40, 0xeeed80, 0xc00002c080)
        C:/hostedtoolcache/windows/go/1.16.3/x64/src/net/http/server.go:1952 +0x8cd
created by net/http.(*Server).Serve
        C:/hostedtoolcache/windows/go/1.16.3/x64/src/net/http/server.go:3013 +0x3b8
```

**Comment by neel1996** [comment]
@Nageswari-droid Fix is ready for #155. Approve the build to pick up the changes.
